modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -292,7 +318,7 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
         else:
             new_columns = groupby_cols.copy()
 
-        new_dtypes = by_frame._dtypes[groupby_cols].tolist()
+        new_dtypes = base._dtypes[groupby_cols].tolist()
```

Comment:
why?

Reply:
`base` frame contains "finalized" grouping and aggregate columns with all of the renaming and possible casts done, so it's more appropriate to get the result dtypes from the `base` rather than from the raw `by_frame`